git-extra/astextplain

```diff
@@ -8,6 +8,9 @@ fi
 
 # XXX output encoding (UTF-8) hardcoded
 case "$1" in
+	*.ods | *.ODS | *odf |*.ODF | *odt | *.ODT)
+		odf2txt "$1" || cat "$1"
```

Comment:
With this example and the repo where to push I can finish this. Maybe need a couple of iterations for details. Let's see.